lib/Dialect/FIRRTL/FIRRTLFieldSource.cpp

```diff
@@ -43,6 +49,13 @@ void FieldSource::visitOp(Operation *op) {
     for (auto &b : r.getBlocks())
       for (auto &op : b)
         visitOp(&op);
+
+  if (op->getNumResults()) {
+    auto type = op->getResult(0).getType();
+    if (dyn_cast<FIRRTLBaseType>(type) &&
+        !cast<FIRRTLBaseType>(type).isGround())
```

Comment:
Since RefType's are root storage when they're ports, is this check because you can't do `isGround` on `FIRRTLType` or because references returned by operations are different?

If the former, consider `getBaseType(type).isGround()`.

Presently this doesn't matter but for something as important as a pointer analysis other transforms will rely on for correctness, I think the bar is high re:being careful and precise, especially given the incoming changes to FIRRTL planned.